src/main/scala/cromwell/engine/WorkflowManager.scala

```diff
@@ -110,6 +120,12 @@ class ActorWorkflowManager extends WorkflowManagerActor {
     } yield context.actorOf(WorkflowActor.buildWorkflowActorProps(id, binding, coercedInputs))
   }
 
+  override def notifyCompletion(obj: Option[AnyRef]): Unit = {
+    obj match {
+      case Some(o) => o.synchronized{o.notify()}
```

Comment:
I'm just skimming at the moment so haven't really parsed what's going on (IOW I'm just lobbing bombs w/o providing suggestions!) but I'd argue in a codebase like this that `synchronized` is approaching anti-pattern status.